src/Worker/Core/Shims/TaskActivityShim.cs

```diff
@@ -11,17 +12,24 @@ namespace Microsoft.DurableTask.Worker.Shims;
 class TaskActivityShim : TaskActivity
 {
     readonly ITaskActivity implementation;
+    readonly ILogger logger;
     readonly DataConverter dataConverter;
     readonly TaskName name;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="TaskActivityShim"/> class.
     /// </summary>
+    /// <param name="loggerFactory">The logger factory.</param>
     /// <param name="dataConverter">The data converter.</param>
     /// <param name="name">The name of the activity.</param>
     /// <param name="implementation">The activity implementation to wrap.</param>
-    public TaskActivityShim(DataConverter dataConverter, TaskName name, ITaskActivity implementation)
+    public TaskActivityShim(
+        ILoggerFactory loggerFactory,
+        DataConverter dataConverter,
+        TaskName name,
+        ITaskActivity implementation)
     {
+        this.logger = Check.NotNull(loggerFactory).CreateLogger(Logs.WorkerCategoryName);
```

Comment:
Maybe the category name here should include _Invocation_ or _Execution_ in it somewhere? As well as _Activity_ or _Activities_ or _TaskActivity_.

`Microsoft.DurableTask.Worker.Invocation.TaskActivity`?

Reply:
I went with simply `Microsoft.DurableTask.Worker.Activities` per the earlier suggestion. I don't see a need to distinguish between logs related to invocations and those which aren't.